backend-api/src/functions/tests/pact/asyncCredential/asyncCredentialConfiguration.ts

```diff
@@ -0,0 +1,68 @@
+import { IAsyncCredentialDependencies } from "../../../asyncCredential/handlerDependencies";
+import { registeredLogs } from "../../../asyncCredential/registeredLogs";
+import { MockEventWriterSuccess } from "../../../services/events/tests/mocks";
+import { Logger } from "../../../services/logging/logger";
+import { MockLoggingAdapter } from "../../../services/logging/tests/mockLogger";
+import {
+  MockClientRegistryServiceGetPartialClientSuccessResultIPV,
+  MockSessionServiceCreateSessionSuccessResult,
+  MockTokenServiceInvalidSignatureIPV,
+  MockTokenServiceSuccessIPV,
+} from "../../../testUtils/asyncCredentialMocks";
+
+const env = {
+  SIGNING_KEY_ID: "mockKid",
+  ISSUER: "mockIssuer",
+  SESSION_TABLE_NAME: "mockTableName",
+  SESSION_TABLE_SUBJECT_IDENTIFIER_INDEX_NAME: "mockIndexName",
+  SESSION_TTL_IN_MILLISECONDS: "12345",
+  SQS_QUEUE: "mockSqsQueue",
+  CLIENT_REGISTRY_PARAMETER_NAME: "mockParmaterName",
+};
+
+export class AsyncCredentialConfiguration {
+  secret: string = "";
+  dependencies: IAsyncCredentialDependencies;
+
+  constructor() {
+    this.dependencies = this.getPassingDependencies();
+  }
+
+  setDependenciesByScenario(scenario?: AsyncCredentialTestScenario) {
+    if (scenario === "badAccessToken is not a valid access token") {
+      this.resetToPassingDependencies();
+      this.dependencies.tokenService = () =>
+        new MockTokenServiceInvalidSignatureIPV();
```

Comment:
I don't think we should be creating a mock for just IPVCore, otherwise it implies we're missing unit test coverage to me?